tools/testsys-config/src/lib.rs

```diff
@@ -0,0 +1,364 @@
+use bottlerocket_variant::Variant;
+pub use error::Error;
+use handlebars::Handlebars;
+use maplit::btreemap;
+use model::SecretName;
+use serde::{Deserialize, Serialize};
+use snafu::ResultExt;
+use std::collections::{BTreeMap, HashMap};
+use std::fs;
+use std::path::Path;
+pub type Result<T> = std::result::Result<T, error::Error>;
+
+/// Configuration needed to run tests
+#[derive(Debug, Default, Deserialize, Serialize, PartialEq, Eq)]
+#[serde(rename_all = "kebab-case")]
+pub struct TestConfig {
+    /// High level configuration for TestSys
+    pub test: Option<Test>,
+
+    #[serde(flatten, serialize_with = "toml::ser::tables_last")]
+    /// Configuration for testing variants
+    pub configs: HashMap<String, GenericConfig>,
+}
+
+impl TestConfig {
+    /// Deserializes a TestConfig from a given path
+    pub fn from_path<P>(path: P) -> Result<Self>
+    where
+        P: AsRef<Path>,
+    {
+        let path = path.as_ref();
+        let test_config_str = fs::read_to_string(path).context(error::FileSnafu { path })?;
+        toml::from_str(&test_config_str).context(error::InvalidTomlSnafu { path })
+    }
+
+    /// Deserializes a TestConfig from a given path, if it exists, otherwise builds a default
+    /// config
+    pub fn from_path_or_default<P>(path: P) -> Result<Self>
```

Comment:
It might be helpful for a user to know (via stdout) whether they are getting their file loaded or getting a default. For example if I have a typo in the path that I pass `/home/me/bottlerocket-teesting/Test.toml`, I could be left scratching my head as to why my configuration file is not working.

Reply:
This issue the same loading behavior as `Infra.toml` if we are going to change one we should probably change both.

Reply:
Hmmm, it should probably warn me there as well. I thought I remembered seeing a warning about a default Infra.toml. That might be because we generate one earlier in the build process when one does not exist.